src/transformers/pipelines/fill_mask.py

```diff
@@ -157,32 +151,40 @@ def __call__(self, *args, targets=None, top_k: Optional[int] = None, **kwargs):
                     values = tf.gather_nd(values, tf.reshape(sort_inds, (-1, 1))).numpy()
                     predictions = target_inds[sort_inds.numpy()]
             else:
-                masked_index = torch.nonzero(input_ids == self.tokenizer.mask_token_id, as_tuple=False)
+                masked_indices = torch.nonzero(input_ids == self.tokenizer.mask_token_id, as_tuple=False)
+                # Fill mask pipeline supports at least one ${mask_token} per sample
+                self.ensure_atleast_one_mask_token(masked_indices.numpy())
 
-                # Fill mask pipeline supports only one ${mask_token} per sample
-                self.ensure_exactly_one_mask_token(masked_index.numpy())
+                logits_multiple = [outputs[i, index.item(), :] for index in masked_indices]
+
+                probs_multiple = [logits.softmax(dim=0) for logits in logits_multiple]
 
-                logits = outputs[i, masked_index.item(), :]
-                probs = logits.softmax(dim=0)
                 if targets is None:
-                    values, predictions = probs.topk(top_k if top_k is not None else self.top_k)
+                    values_all = []
+                    predictions_all = []
+                    for probs in probs_multiple:
+                        values, predictions = probs.topk(top_k if top_k is not None else self.top_k)
+                        values_all.append(values)
+                        predictions_all.append(predictions)
                 else:
+                    # pending for when the target tokens are specifically input to the model.
                     values = probs[..., target_inds]
                     sort_inds = list(reversed(values.argsort(dim=-1)))
                     values = values[..., sort_inds]
                     predictions = target_inds[sort_inds]
 
-            for v, p in zip(values.tolist(), predictions.tolist()):
+            for i, item in enumerate(values_all[0]):
```

Comment:
Try to avoid all indices if possible in your loops.

It's usually a code smell (not necessarily but it's most likely avoidable here).
It will also likely lead to indexing errors at some point.

Reply:
I see, yeah I think I can get rid of this by zipping.